lua/nvim-treesitter/highlight.lua

```diff
@@ -6,116 +6,229 @@ local configs = require "nvim-treesitter.configs"
+    ["@float"] = "TSFloat",
 
-hlmap["symbol"] = "TSSymbol"
+    ["@function"] = "TSFunction",
+    ["@function.call"] = "TSFunctionCall",
+    ["@function.builtin"] = "TSFuncBuiltin",
+    ["@function.macro"] = "TSFuncMacro",
 
-hlmap["tag"] = "TSTag"
-hlmap["tag.attribute"] = "TSTagAttribute"
-hlmap["tag.delimiter"] = "TSTagDelimiter"
+    ["@include"] = "TSInclude",
 
-hlmap["text"] = "TSText"
-hlmap["text.strong"] = "TSStrong"
-hlmap["text.emphasis"] = "TSEmphasis"
-hlmap["text.underline"] = "TSUnderline"
-hlmap["text.strike"] = "TSStrike"
-hlmap["text.title"] = "TSTitle"
-hlmap["text.literal"] = "TSLiteral"
-hlmap["text.uri"] = "TSURI"
-hlmap["text.math"] = "TSMath"
-hlmap["text.reference"] = "TSTextReference"
-hlmap["text.environment"] = "TSEnvironment"
-hlmap["text.environment.name"] = "TSEnvironmentName"
+    ["@keyword"] = "TSKeyword",
+    ["@keyword.function"] = "TSKeywordFunction",
+    ["@keyword.operator"] = "TSKeywordOperator",
+    ["@keyword.return"] = "TSKeywordReturn",
 
-hlmap["text.note"] = "TSNote"
-hlmap["text.warning"] = "TSWarning"
-hlmap["text.danger"] = "TSDanger"
+    ["@label"] = "TSLabel",
 
-hlmap["todo"] = "TSTodo"
+    ["@method"] = "TSMethod",
+    ["@method.call"] = "TSMethodCall",
 
-hlmap["type"] = "TSType"
-hlmap["type.builtin"] = "TSTypeBuiltin"
-hlmap["type.qualifier"] = "TSTypeQualifier"
-hlmap["type.definition"] = "TSTypeDefinition"
+    ["@namespace"] = "TSNamespace",
 
-hlmap["variable"] = "TSVariable"
-hlmap["variable.builtin"] = "TSVariableBuiltin"
+    ["@none"] = "TSNone",
+    ["@number"] = "TSNumber",
+
+    ["@operator"] = "TSOperator",
+
+    ["@parameter"] = "TSParameter",
+    ["@parameter.reference"] = "TSParameterReference",
+
+    ["@preproc"] = "TSPreProc",
+
+    ["@property"] = "TSProperty",
+
+    ["@punctuation.delimiter"] = "TSPunctDelimiter",
+    ["@punctuation.bracket"] = "TSPunctBracket",
+    ["@punctuation.special"] = "TSPunctSpecial",
+
+    ["@repeat"] = "TSRepeat",
+
+    ["@storageclass"] = "TSStorageClass",
+
+    ["@string"] = "TSString",
+    ["@string.regex"] = "TSStringRegex",
+    ["@string.escape"] = "TSStringEscape",
+    ["@string.special"] = "TSStringSpecial",
+
+    ["@symbol"] = "TSSymbol",
+
+    ["@tag"] = "TSTag",
+    ["@tag.attribute"] = "TSTagAttribute",
+    ["@tag.delimiter"] = "TSTagDelimiter",
+
+    ["@text"] = "TSText",
+    ["@text.strong"] = "TSStrong",
+    ["@text.emphasis"] = "TSEmphasis",
+    ["@text.underline"] = "TSUnderline",
+    ["@text.strike"] = "TSStrike",
+    ["@text.title"] = "TSTitle",
+    ["@text.literal"] = "TSLiteral",
+    ["@text.uri"] = "TSURI",
+    ["@text.math"] = "TSMath",
+    ["@text.reference"] = "TSTextReference",
+    ["@text.environment"] = "TSEnvironment",
+    ["@text.environment.name"] = "TSEnvironmentName",
+
+    ["@text.note"] = "TSNote",
+    ["@text.warning"] = "TSWarning",
+    ["@text.danger"] = "TSDanger",
+
+    ["@todo"] = "TSTodo",
+
+    ["@type"] = "TSType",
```

Comment:
should the 0.8 branch really do this for captures that now will be handled in core, like `@type`, `@string` and so on? (while nvim-treesitter still handling its own groups, like `@text.emphasis` , of course)

Reply:
There is no 0.8 branch...

But yes, this is purely a compatibility shim; after 0.8 is released with core highlighting, we'll have to revisit the whole thing and simplify drastically.

(There may still be a good reason to keep the now well-established `TSFoo` highlight groups around, for the sake of color schemes -- better to have all if we want to keep some.)

Reply:
Ok, so at a later point (after 0.8), this entire section could be made opt-in rather than opt-out (or removed entirely, provided there is another solution for the special groups that are regardless needed).

Reply:
Indeed; that is a strategic decision on how far the `TSFoo` groups are considered "branding".